File: UI_Blazor_WebApp/UI_Blazor_WebApp/Services/FinanceRecordsService.cs
using UI_Blazor_WebApp.Models;

namespace UI_Blazor_WebApp.Services
{
    public class FinanceRecordsService
    {
        private readonly HttpClient httpClient;

        private const string financeRecordsRoute = "/finance/records";

        public FinanceRecordsService(HttpClient httpClient)
        {
            this.httpClient = httpClient;
        }

        public async Task<IEnumerable<FinanceRecordModel>> GetAllFinanceRecords()
        {
            var financeRecords = await httpClient.GetFromJsonAsync<FinanceRecordModel[]>(financeRecordsRoute);

            if (financeRecords == null)
            {
                financeRecords = Array.Empty<FinanceRecordModel>();
            }

            return financeRecords.ToList();
        }

        public async Task PostFinanceRecord(FinanceRecordModel record)
        {
            await httpClient.PostAsJsonAsync(financeRecordsRoute, record);
        }

        public async Task DeleteFinanceRecord(Guid recordId)
        {
            await httpClient.DeleteAsync($"{financeRecordsRoute}/{recordId}");
        }
    }
}

File: UI_Angular/src/app/services/finance-records.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, map } from 'rxjs';

import { IFinanceRecord } from '../models/financeRecord';

export const financeApiHost = 'https://localhost:8080';
export const financeRecordsRoute = '/finance/records';

@Injectable({
  providedIn: 'root'
})
export class FinanceRecordsService {

  constructor(private httpClient: HttpClient) { }

  getAllFinanceRecords(): Observable<IFinanceRecord[]> {
    return this.httpClient
      .get<IFinanceRecord[]>(`${financeApiHost}${financeRecordsRoute}`)
      .pipe(
        map((records: IFinanceRecord[]) => records.map((record: IFinanceRecord) => (
          {
            ...record,
            date: new Date(record.date),
          }
        ))),
      );
  }

  postFinanceRecord(record: IFinanceRecord): Observable<Object> {
    return this.httpClient.post(`${financeApiHost}${financeRecordsRoute}`, record);
  };

  deleteFinanceRecord = (recordId: string): Observable<Object> => {
    return this.httpClient.delete(`${financeApiHost}${financeRecordsRoute}/${recordId}`);
  };
}
